### ISSUE TITLE
JFYI - Windows builds are broken, build system issue, ignore

### ISSUE THREAD

**Issue by MSP-Greg** [puma/puma#2040]
Today some files/packages in the MSYS2 build system were updated, and builds on both Appveyor & GH Actions are broken. ruby/ruby, ruby-loco, here, etc...

I've been using MSYS2 to build Ruby since Ruby 2.4 was trunk, and I don't recall any failures of this type.  So, it's not common, and hopefully the issue can get fixed soon...

**Comment by MSP-Greg** [comment]
The issue is an MSYS2 package built on code from Mingw-w64, as much of the toolchain code is from gcc.gnu.org.

Anyway, I adjusted the msys2-action code to patch around it, while still using the updated gcc 9.2.0...

Once they fix the issue, I'll revert (code is at https://github.com/MSP-Greg/msys2-action)

**Comment by nateberkopec** [comment]
Is there an open issue on their windows repo?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by MSP-Greg** [comment]
Sorry, who is 'their'?

**Comment by MSP-Greg** [comment]
Today, with some help from the people in MSYS2, I sorted this out.  PR https://github.com/puma/puma/pull/2044 needs to be accepted for windows builds to compile.

There is still an issue with Ruby, see
https://bugs.ruby-lang.org/issues/16265

JFYI, I've been active in Windows Ruby for quite while, and have also been involved in fixes to the Ruby test suites.  Often, the MSYS2 Windows issues are really 'new' gcc issues, as mentioned in the PR.

This repo (and others) use a 'GitHub action' of mine, and I updated it this morning, which is why there's breakage here.

The update fully updates the MSYS2 system, while the interim fix left some libraries in the build system at version that just happened to be in place on the old Appveyor images.  Anyone installing a new version of MSYS2 can't install those files using pacman without downloading packages from elsewhere and installing them locally.  IOW, not normal operations.

Sorry, but sometimes I get tired of explaining Windows issues...
